webrtc/RTCPeerConnection-set-description.html

```diff
@@ -0,0 +1,395 @@
+<!doctype html>
+<title>RTCPeerConnection Set Description tests</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script type="text/javascript">
+  'use strict';
+
+  const audioLineRegex = /\r\nm=audio.+\r\n/g
+  const videoLineRegex = /\r\nm=video.+\r\n/g
+
+  function countAudioLine(sdp) {
+    const matches = sdp.match(audioLineRegex);
+    if(matches === null) {
+      return 0;
+    } else {
+      return matches.length;
+    }
+  }
+
+  function countVideoLine(sdp) {
+    const matches = sdp.match(videoLineRegex);
+    if(matches === null) {
+      return 0;
+    } else {
+      return matches.length;
+    }
+  }
+
+  function similarMediaDescriptions(sdp1, sdp2) {
+    if(sdp1 === sdp2) {
+      return true;
+    } else if(countAudioLine(sdp1) !== countAudioLine(sdp2)) {
+      return false;
+    } else if(countVideoLine(sdp1) !== countVideoLine(sdp2)) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  function assert_defined(object) {
+    if(object === undefined || object === null)
+      assert_unreached('object must not be undefined or null');
+  }
+
+  function isSimilarSdp(sdp1, sdp2) {
+    assert_defined(sdp1);
+    assert_defined(sdp2);
+
+    if(sdp1.type !== sdp2.type) {
+      return false;
+    } else {
+      // We can't do string comparison to the SDP content,
+      // because RTCPeerConnection may return SDP that is
+      // slightly modified from what is given to it due to
+      // ice candidate events or serialization.
+      // To hack around this, if the SDP strings are not
```

Comment:
This seems a bit risky. Could you instead explicitly remove the bits that the spec describes can change, and assert that nothing else does?